src/index.ts

```diff
@@ -202,19 +203,32 @@ export default declare((api: any, options: PluginOptions) => {
             },
 
             // `interface FooProps {}`
-            TSInterfaceDeclaration({ node }: Path<t.TSInterfaceDeclaration>) {
+            // @ts-ignore
+            TSInterfaceDeclaration(path: Path<t.TSInterfaceDeclaration>) {
+              const { node } = path;
+
               state.componentTypes[node.id.name] = extractTypeProperties(
                 node,
                 state.componentTypes,
               );
+
+              if (state.options.declarePropTypeVariables) {
+                addVariable(path, node.id.name, state);
```

Comment:
We should only add variables when the interface is for props/state. We shouldn't do it for *every* interface.

Reply:
How we can do it? Match with the name suffix `Props|State`?

Reply:
You'll need to use the extracted generic names from classes/functions, and cross reference that.

Reply:
What if we don't have classes/functions in `d.ts`?)

Reply:
Or we should write everytime classes/function in `d.ts`?

Reply:
I won't allow this feature if it generates for every interface/type alias that exists. That would blow up the filesize unnecessarily.

Maybe if the setting was a regex pattern:

```
declarePropTypesVariables: /(Props|State)$/,
```

Reply:
I will think it.